module/spl/spl-taskq.c

```diff
@@ -1063,6 +1072,8 @@ taskq_create(const char *name, int nthreads, pri_t pri,
 
 	/* Wait for all threads to be started before potential destroy */
 	wait_event(tq->tq_wait_waitq, tq->tq_nthreads == count);
+	/* reset spawning count to zero */
```

Comment:
Maybe?:

    /* revert the changes to tq_nspawn adjustment in taskq_thread() */